examples/seismic/tutorials/05_dask.ipynb

```diff
@@ -0,0 +1,496 @@
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "from scipy import optimize\n",
+    "\n",
+    "# Define bounding box constraints on the solution.\n",
+    "def apply_box_constraint(m):\n",
+    "    # Maximum accepted velocity is 3.5 km/sec (true is 3 km/sec)\n",
+    "    # Minimum accepted velocity is 2 km/sec (true is 2.5 km/sec)\n",
+    "    return np.clip(m, 0.08, 0.25)\n",
+    "\n",
+    "# Many methods in scipy.optimize.minimize allow us to define a callback\n",
+    "# function that can operate on the solution after every iteration. Here\n",
+    "# we use this to apply box constraints and to monitor the true relative\n",
+    "# solution error.\n",
+    "relative_error = []\n",
+    "def fwi_callbacks(x):\n",
+    "    # Apply boundary constraint\n",
+    "    x.data[:] = apply_box_constraint(x)\n",
+    "    \n",
+    "    # Calculate true relative error\n",
+    "    true_x = get_true_model().m.data.flatten()\n",
+    "    relative_error.append(np.linalg.norm((x-true_x)/true_x))\n",
+    "\n",
+    "def fwi(model, param, ftol=0.1, maxiter=20):\n",
+    "    result = optimize.minimize(fwi_gradient,\n",
+    "                               model.m.data.flatten().astype(np.float64),\n",
+    "                               args=(param, ), method='L-BFGS-B', jac=True,\n",
+    "                               callback=fwi_callbacks,\n",
+    "                               options={'ftol':ftol,\n",
+    "                                        'maxiter':maxiter,\n",
+    "                                        'disp':True})\n",
+    "\n",
+    "    return result"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We new apply our FWI function and have a look at the result."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "      fun: 59.130213695617584\n",
+      " hess_inv: <32761x32761 LbfgsInvHessProduct with dtype=float64>\n",
+      "      jac: array([ 0.,  0.,  0., ...,  0.,  0.,  0.])\n",
+      "  message: b'CONVERGENCE: REL_REDUCTION_OF_F_<=_FACTR*EPSMCH'\n",
+      "     nfev: 15\n",
+      "      nit: 14\n",
+      "   status: 0\n",
+      "  success: True\n",
+      "        x: array([ 0.16,  0.16,  0.16, ...,  0.16,  0.16,  0.16])\n"
+     ]
+    }
+   ],
+   "source": [
+    "#NBVAL_SKIP\n",
+    "\n",
+    "# Adjust the level of diagnostic output from Devito\n",
+    "from devito import configuration\n",
+    "configuration['log_level'] = 'WARNING'\n",
+    "\n",
+    "# Set up inversion parameters.\n",
+    "param = {'t0': 0.,\n",
+    "         'tn': 1000.,              # Simulation last 1 second (1000 ms)\n",
+    "         'f0': 0.010,              # Source peak frequency is 10Hz (0.010 kHz)\n",
+    "         'nshots': 16}             # Number of shots to create gradient from\n",
```

Comment:
`nshots` seems to be different in different places. Either use consistently or explain why it is different.

Reply:
Explained that we can now process more shots because of parallel.